src/Features/CSharp/Portable/CodeFixes/Nullable/CSharpDeclareAsNullableCodeFixProvider.cs

```diff
@@ -0,0 +1,152 @@
+            var node = root.FindNode(diagnostic.Location.SourceSpan, getInnermostNodeForTie: true);
+
+            var declarationTypeToFix = TryGetDeclarationTypeToFix(node);
+            if (declarationTypeToFix == null)
+            {
+                return;
+            }
+
+            context.RegisterCodeFix(new MyCodeAction(
+                c => FixAsync(context.Document, diagnostic, c)),
+                context.Diagnostics);
+        }
+
+        protected override Task FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics,
+            SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            var root = editor.OriginalRoot;
+
+            // a method can have multiple `return null;` statements, but we should only fix its return type once
+            var alreadyHandled = PooledHashSet<TypeSyntax>.GetInstance();
+
+            foreach (var diagnostic in diagnostics)
+            {
+                var node = diagnostic.Location.FindNode(getInnermostNodeForTie: true, cancellationToken);
+                MakeDeclarationNullable(document, editor, node, alreadyHandled);
+            }
+
+            alreadyHandled.Free();
+            return Task.CompletedTask;
+        }
+
+        private static void MakeDeclarationNullable(Document document, SyntaxEditor editor, SyntaxNode node, HashSet<TypeSyntax> alreadyHandled)
+        {
+            var declarationTypeToFix = TryGetDeclarationTypeToFix(node);
+            if (declarationTypeToFix != null && alreadyHandled.Add(declarationTypeToFix))
+            {
+                var fixedDeclaration = SyntaxFactory.NullableType(declarationTypeToFix.WithoutTrivia()).WithTriviaFrom(declarationTypeToFix);
+                editor.ReplaceNode(declarationTypeToFix, fixedDeclaration);
+            }
+        }
+
+        private static TypeSyntax TryGetDeclarationTypeToFix(SyntaxNode node)
+        {
+            if (!node.IsKind(SyntaxKind.NullLiteralExpression))
+            {
+                return null;
+            }
+
+            if (node.IsParentKind(SyntaxKind.ReturnStatement))
+            {
+                var containingMember = node.GetAncestors().FirstOrDefault(a => a.IsReturnableConstruct());
+
+                if (containingMember == null)
+                {
+                    return null;
+                }
+
+                switch (containingMember.Kind())
+                {
+                    case SyntaxKind.MethodDeclaration:
+                        // string M() { return null; }
+                        return ((MethodDeclarationSyntax)containingMember).ReturnType;
+
+                    case SyntaxKind.PropertyDeclaration:
+                        // string x { get { return null; } }
+                        return ((PropertyDeclarationSyntax)containingMember).Type;
+
+                    default:
+                        return null;
```

Comment:
you could go back and remove `IsReturnableConstruct` or I would persnally keep it and use pattern matching here:
```c#
case MethodDeclarationSyntax method: ...
case AccessorDeclarationSyntax accessor when accessor.Parent is PropertyDeclarationSyntax property: ...
```

Reply:
this make me think, what about indexers?